src/OpenFeature/Api.cs

```diff
@@ -38,25 +39,48 @@ public sealed class Api : IEventBus
         private Api() { }
 
         /// <summary>
-        /// Sets the feature provider. In order to wait for the provider to be set, and initialization to complete,
+        /// Sets the default feature provider to given clientName without awaiting its initialization.
+        /// </summary>
+        /// <remarks>The provider cannot be set to null. Attempting to set the provider to null has no effect.</remarks>
+        /// <param name="featureProvider">Implementation of <see cref="FeatureProvider"/></param>
+        [Obsolete("Will be removed in later versions; use SetProviderAsync, which can be awaited")]
+        public void SetProvider(FeatureProvider featureProvider)
+        {
+            this.EventExecutor.RegisterDefaultFeatureProvider(featureProvider);
+            this._repository.SetProvider(featureProvider, this.GetContext()).ConfigureAwait(false);
+        }
+
+        /// <summary>
+        /// Sets the default feature provider. In order to wait for the provider to be set, and initialization to complete,
         /// await the returned task.
         /// </summary>
         /// <remarks>The provider cannot be set to null. Attempting to set the provider to null has no effect.</remarks>
         /// <param name="featureProvider">Implementation of <see cref="FeatureProvider"/></param>
-        public async Task SetProvider(FeatureProvider featureProvider)
+        public async Task SetProviderAsync(FeatureProvider featureProvider)

```

Comment:
While we're revisiting this...

I have some work planned/started (see: #185) to go through and `ValueTask`-ize some (all?) of our public async methods that have a ~~reasonable~~ vague likelihood of either completing synchronously, and/or being likely to be implemented/plugged into from contrib/user code in ways that could benefit from some non-allocating completions.

```suggestion
        public ValueTask SetProviderAsync(FeatureProvider featureProvider)
```

Reply:
I can see the benefit of using `ValueTask` for evaluations, but I'm not sure I do here. From the .NET 8 [API docs](https://learn.microsoft.com/en-us/dotnet/api/system.threading.tasks.valuetask?view=net-8.0):

> the default choice for any asynchronous method that does not return a result should be to return a [Task](https://learn.microsoft.com/en-us/dotnet/api/system.threading.tasks.task?view=net-8.0). **Only if performance analysis proves it worthwhile should a ValueTask be used instead of a [Task](https://learn.microsoft.com/en-us/dotnet/api/system.threading.tasks.task?view=net-8.0)**.

Setting a provider is not on a "hot path" and it's potentially used in more complex logic where the additional utilities of a plain Task might be helpful.

Reply:
I'm interested in what @benjiro and @kinyoklion think about using a `ValueTask` here.

Reply:
Just to add: https://devblogs.microsoft.com/dotnet/understanding-the-whys-whats-and-whens-of-valuetask/

Honestly, I don't see much of a gain here. I don't see setting a provider to be a high-performance bottleneck, especially ideally you set the provider during the application startup.

Reply:
> I can see the benefit of using ValueTask for flag evaluations and provider resolvers, but I'm not sure I do here. From the .NET 8 [API docs](https://learn.microsoft.com/en-us/dotnet/api/system.threading.tasks.valuetask?view=net-8.0):
>
>> [...]
>
> Setting a provider is not on a "hot path" and it's potentially used in more complex logic where the additional utilities of a plain Task might be helpful.

That's a good point, especially considering that, while I'm a fan of supporting multi-providers, the most common steady-state (i.e. outside of slow, in-place migrations) will most likely see apps with a singular provider.

I'm happy to hear we're open to `ValueTask` on other, more obvious, hot-paths, so I work with the stance that `we don't consider async init a hot path`, if that's where the consensus ends up.

The only context I want to add is that I read `[...] default choice for any [...]` as more so speaking caution to consumer apps (e.g. don't make every method in your MVC web app return `ValueTask`) vs to library/SDK authors seeking to be as low-impact as possible.

So, if I _were_ going to make the argument in favor of `ValueTask` in our async init logic, it would be that these paths might not be hot, but in most uses, they'll probably precede all other app logic, and being lightweight on hosting startup can be nice in some (arguably niche) use cases (e.g. cold start lambdas, CLI apps, etc).

_(If anyone actually wants takes this^ "nice use cases" argument seriously, we could write benchmarks to see if it even matters, but tbh, I'm not sure I'm even taking it that seriously after having written it out lol.)_

Reply:
`ValueTask` does have merit, and I would be open to it for some demonstrable cases, I don't think this would be a case considerable improved by it. Considering it is a matter of allocations, and that `ValueTask` isn't flatly fewer allocations, it would make sense for high frequency use cases where we suspect they will usually be able to complete synchronously (if they are actually consistently async, then it probably doesn't help). This seems reasonable for evaluations where most of the time, most providers, would be satisfying the request from cache.